Styling-fixes-for-ui-frameworks

Aby-JS opened the pull request.

Aby-JS committed:
fixed phone input heigh issue for tailwind css

Aby-JS committed:
Publish

 - @corbado/react@2.6.1-alpha.0
 - @corbado/shared-ui@2.6.1-alpha.0
 - @corbado/types@2.4.1-alpha.0
 - @corbado/web-core@2.4.1-alpha.0
 - @corbado/web-js@2.6.1-alpha.0

Aby-JS committed:
updated react example application

Aby-JS committed:
updated other example version

incorbador commented on packages/shared-ui/src/styles/index.css:
```diff
@@ -261,7 +261,7 @@
   flex: 4;
   border: 0px;
-  height: var(--cb-input-field-height);
+  height: calc(var(--cb-input-field-height) - 0.2rem);
```
Is this a fix to the problem we were discussing? (that CSS frameworks like tailwind override styles for HTML elements like input, div, ...) => I thought we have to explicitly set some CSS for elements where Tailwind defines defaults that don't work with out stylings (the change above looks more like a quick fix to me though)

Aby-JS replied:
Basically the design of our phone input is different then other input fields.
For other identifiers our container is of `input` type and has a height of `var(--cb-input-field-height)`. Also, inputs have borders.
For phone input our container is `div` since it has two components a selection dropdown button and an input container to add remaining phone number digits. In this scenario the div has borders and input borderless. Though the issue became prominent with tailwind's stylings, I think that this should also be how it should be implemented. Since the parent has borders the child should have a slightly smaller height then parent.

incorbador commented:
In addition to the phone input issue that we already address in this PR, we have an issue with the social buttons. In dev panel our social buttons look like this:
![image](https://github.com/corbado/javascript/assets/134682658/9e5e530c-6a16-44e8-9cf0-06055bc4a042)

In playground they look like this:
![image](https://github.com/corbado/javascript/assets/134682658/1c2b4372-d700-4df5-a5c3-8d35183939e7)

=> in playground they are centered vertically, but in dev panel they are not (we had to apply a custom style customization to fix this: padding-bottom: .1rem;) 
=> the question here is: why do we need the style customization in the dev panel and how can we avoid it?

Aby-JS committed:
fixed input styling issues

Aby-JS committed:
Publish

 - @corbado/react@2.6.2-alpha.0
 - @corbado/shared-ui@2.6.2-alpha.0
 - @corbado/web-js@2.6.2-alpha.0

Aby-JS committed:
updated styling fixes

Aby-JS committed:
fixed vuetify icon alignment issue

Aby-JS committed:
Publish

 - @corbado/react@2.7.0-alpha.0
 - @corbado/shared-ui@2.7.0-alpha.0
 - @corbado/web-js@2.7.0-alpha.0

Aby-JS committed:
fixed web-js build issues

Aby-JS committed:
Merge branch 'develop' into styling-fixes-for-ui-frameworks
